src/main/java/com/microfocus/application/automation/tools/octane/CIJenkinsServicesImpl.java

```diff
@@ -101,497 +104,537 @@
+            } else {
+                result = createPipelineNodeFromJobName(item.getFullName());
+                if (item.getClass().getName().equals(JobProcessorFactory.WORKFLOW_MULTI_BRANCH_JOB_NAME)) {
+                    result.setMultiBranchType(MultiBranchType.MULTI_BRANCH_PARENT);
+                }
+            }
+            return result;
+        } finally {
+            stopImpersonation(securityContext);
+        }
+    }
+
+    @Override
+    public void runPipeline(String jobCiId, CIParameters ciParameters) {
+        ACLContext securityContext = startImpersonation();
+        try {
+            Job job = getJobByRefId(jobCiId);
+            if (job != null) {
+                if (job instanceof AbstractProject && ((AbstractProject) job).isDisabled()) {
+                    //disabled job is not runnable and in this context we will handle it as 404
+                    throw new ConfigurationException(HttpStatus.SC_NOT_FOUND);
+                }
+                boolean hasBuildPermission = job.hasPermission(Item.BUILD);
+                if (!hasBuildPermission) {
+                    stopImpersonation(securityContext);
+                    throw new PermissionException(HttpStatus.SC_FORBIDDEN);
+                }
+                if (job instanceof AbstractProject || job.getClass().getName().equals(JobProcessorFactory.WORKFLOW_JOB_NAME)) {
+                    doRunImpl(job, ciParameters);
+                }
+            } else {
+                throw new ConfigurationException(HttpStatus.SC_NOT_FOUND);
+            }
+        } finally {
+            stopImpersonation(securityContext);
+        }
+    }
+
+    @Override
+    public void stopPipelineRun(String jobCiId, CIParameters ciParameters) {
+        ACLContext securityContext = startImpersonation();
+        try {
+            Job job = getJobByRefId(jobCiId);
+            if (job != null) {
+                boolean hasAbortPermissions = job.hasPermission(Item.CANCEL);
+                if (!hasAbortPermissions) {
+                    stopImpersonation(securityContext);
+                    throw new PermissionException(HttpStatus.SC_FORBIDDEN);
+                }
+                if (job instanceof AbstractProject || job.getClass().getName().equals(JobProcessorFactory.WORKFLOW_JOB_NAME)) {
+                    doStopImpl(job, ciParameters);
+                }
+            } else {
+                throw new ConfigurationException(HttpStatus.SC_NOT_FOUND);
+            }
+        } finally {
+            stopImpersonation(securityContext);
+        }
+    }
+
+    @Override
+    public CIBranchesList getBranchesList(String jobCiId, String filterBranchName) {
+        ACLContext securityContext = startImpersonation();
+        List<Branch> result = new ArrayList<>();
+        try {
+            Item item = Jenkins.get().getItemByFullName(jobCiId);
+            if (item != null) {
+                boolean hasAbortPermissions = item.hasPermission(Item.CANCEL);
```

Comment:
why to you need to check if user have cancel permission? I think it relevant only for stop pipeline run action